com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs

```diff
@@ -143,27 +142,37 @@ void GeneratePlaneMesh(Mesh theMesh, IReadOnlyList<Vector3> positions, IReadOnly
 
         // Generates a procedural mesh composed of Lines (normals, uvs). Normals for lines are ill defined (1 free DOF ), so they are in fact tangents alone the line.
         // If velocities are present, they are bound to TEXCOORD5
-        void GenerateLineMesh(Mesh theMesh, Vector3[] positionsM, int[] curvesCountsM)
+        void GenerateLineMesh(Mesh theMesh, Vector3[] positionsM, int[] curveOffsetM)
         {
-            var curveCount = curvesCountsM.Length;
-            var indexArraySize = (positionsM.Length - curvesCountsM.Length) * 2;
+            var curveCount = curveOffsetM.Length;
+            var indexArraySize = (positionsM.Length - curveOffsetM.Length) * 2;
 
 
-            using (var curveCounts = new NativeArray<int>(curvesCountsM.Length, Allocator.TempJob))
+            using (var curveCounts = new NativeArray<int>(curveOffsetM.Length, Allocator.TempJob))
             using (var vertices = new NativeArray<Vector3>(positionsM.Length, Allocator.TempJob))
-            using (var strideArray = new NativeArray<int>(curvesCountsM.Length, Allocator.TempJob))
+            using (var strideArray = new NativeArray<int>(curveOffsetM.Length, Allocator.TempJob))
             using (var particleTangent = new NativeArray<Vector3>(positionsM.Length, Allocator.TempJob))
             using (var particleUV = new NativeArray<Vector2>(positionsM.Length, Allocator.TempJob))
             using (var indices = new NativeArray<int>(indexArraySize, Allocator.TempJob))
             {
                 vertices.CopyFrom(positionsM);
-                curveCounts.CopyFrom(curvesCountsM);
-                var cnt = 0;
-                for (var i = 0; i < strideArray.Length; ++i)
+                strideArray.CopyFrom(curveOffsetM);
+
+                if (curveOffsetM.Length > 1)
```

Comment:
Not sure I understand what's happening here with the offset length check.

Reply:
Will do comment